### ISSUE TITLE
Vulkan: Black screen on Android devices that don't support the `VK_HDR_create_renderpass2` extension

### ISSUE THREAD

**Issue by norrath-hero** [godotengine/godot#67146]
### Godot version

Godot Engine v4.0.beta2.official

### System information

VIVO Z3, Android 9, Snapdragon 710AIE, Vulkan API 1.1.0 - Using Vulkan Device #0: Qualcomm - Adreno (TM) 616

### Issue description

Can't run a simple csg scene with all 3 types light enabled!
![image](https://user-images.githubusercontent.com/73905273/194769197-cc7e97a2-19ae-4f59-911a-8cd76e5b032b.png)

![image](https://user-images.githubusercontent.com/73905273/194831411-8cffc634-375e-449b-b5ca-32905a86f545.png)

adb logcat

```
10-10 00:35:21.026  7391  7433 E godot   : USER ERROR: Condition "err" is true. Returning: ERR_CANT_CREATE
10-10 00:35:21.026  7391  7433 E godot   :    at: _update_swap_chain (drivers/vulkan/vulkan_context.cpp:1864) - Condition "err" is true. Returning: ERR_CANT_CREATE
10-10 00:35:21.026  7391  7433 E godot   : USER ERROR: Condition "err != OK" is true. Returning: ERR_CANT_CREATE
10-10 00:35:21.026  7391  7433 E godot   :    at: _window_create (drivers/vulkan/vulkan_context.cpp:1479) - Condition "err != OK" is true. Returning: ERR_CANT_CREATE
10-10 00:35:21.043  7391  7433 E godot   : USER ERROR: Format 'R8_Uint' does not support usage as storage image.
10-10 00:35:21.043  7391  7433 E godot   :    at: texture_create (drivers/vulkan/rendering_device_vulkan.cpp:1831) - Method/function failed. Returning: RID()
10-10 00:35:21.105  2746  2746 I NetLowlatency: setDataPhoneId: 0
10-10 00:35:21.105  2746  2746 I NetLowlatency: is automode : true
10-10 00:35:21.105  2746  2746 I NetLowlatency: get the current uplink latency level is 1
10-10 00:35:21.145   721  3434 W SurfaceFlinger: Attempting to set client state on removed layer: Splash Screen org.godotengine.demo#0
10-10 00:35:21.145   721  3434 W SurfaceFlinger: Attempting to destroy on removed layer: Splash Screen org.godotengine.demo#0
10-10 00:35:21.110  1604  4212 I chatty  : uid=1000(system) Binder:1604_18 identical 1 line
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:21.848  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:21.848  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:21.848  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   :    at: render_pipeline_create (drivers/vulkan/rendering_device_vulkan.cpp:6449) - Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:21.848  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   :    at: render_pipeline_create (drivers/vulkan/rendering_device_vulkan.cpp:6449) - Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:21.848  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   :    at: render_pipeline_create (drivers/vulkan/rendering_device_vulkan.cpp:6449) - Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:21.848  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   :    at: render_pipeline_create (drivers/vulkan/rendering_device_vulkan.cpp:6449) - Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:21.848  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.848  7391  7433 E godot   :    at: render_pipeline_create (drivers/vulkan/rendering_device_vulkan.cpp:6449) - Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:21.850  2139  2139 I chatty  : uid=10023(com.android.systemui) identical 2 lines
10-10 00:35:21.855   721  3434 D BufferLayer: triger signalLayerUpdate

...

10-10 00:35:22.151  7391  7433 E godot   :    at: framebuffer_create_multipass (drivers/vulkan/rendering_device_vulkan.cpp:4170) - Condition "!size_set" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: All attachments unused.
10-10 00:35:22.151  7391  7433 E godot   :    at: framebuffer_create_multipass (drivers/vulkan/rendering_device_vulkan.cpp:4170) - Condition "!size_set" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: All attachments unused.
10-10 00:35:22.151  7391  7433 E godot   :    at: framebuffer_create_multipass (drivers/vulkan/rendering_device_vulkan.cpp:4170) - Condition "!size_set" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Attempted to name invalid ID: 0
10-10 00:35:22.151  7391  7433 E godot   :    at: set_resource_name (drivers/vulkan/rendering_device_vulkan.cpp:8902) - Attempted to name invalid ID: 0

...

10-10 00:35:22.151  7391  7433 E godot   :    at: framebuffer_create_multipass (drivers/vulkan/rendering_device_vulkan.cpp:4170) - Condition "!size_set" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Attempted to name invalid ID: 0
10-10 00:35:22.151  7391  7433 E godot   :    at: set_resource_name (drivers/vulkan/rendering_device_vulkan.cpp:8902) - Attempted to name invalid ID: 0
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: All attachments unused.
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: set_resource_name (drivers/vulkan/rendering_device_vulkan.cpp:8902) - Attempted to name invalid ID: 0
10-10 00:35:22.151  7391  7433 E godot   :    at: framebuffer_create_multipass (drivers/vulkan/rendering_device_vulkan.cpp:4170) - Condition "!size_set" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.151  7391  7433 E godot   :    at: framebuffer_create_multipass (drivers/vulkan/rendering_device_vulkan.cpp:4170) - Condition "!size_set" is true. Returning: RID()
10-10 00:35:22.152  7391  7433 E godot   : USER ERROR: Attempted to name invalid ID: 0
10-10 00:35:22.152  7391  7433 E godot   : USER ERROR: Attempted to name invalid ID: 0
10-10 00:35:22.152  7391  7433 E godot   : USER ERROR: All attachments unused.
10-10 00:35:22.152  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.152  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.152  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.152  7391  7433 E godot   :    at: texture_create_shared_from_slice (drivers/vulkan/rendering_device_vulkan.cpp:2273) - Condition "!src_texture" is true. Returning: RID()
10-10 00:35:22.152  7391  7433 E godot   : USER ERROR: Attempted to name invalid ID: 0
10-10 00:35:22.152  7391  7433 E godot   : USER ERROR: Condition "!src_texture" is true. Returning: RID()

...

10-10 00:35:22.210  7391  7433 E godot   : USER ERROR: Condition "!framebuffer" is true. Returning: INVALID_ID
10-10 00:35:22.211  7391  7433 E godot   :    at: framebuffer_format_get_texture_samples (drivers/vulkan/rendering_device_vulkan.cpp:4081) - Condition "!E" is true. Returning: TEXTURE_SAMPLES_1
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:22.211  7391  7433 E godot   :    at: render_pipeline_create (drivers/vulkan/rendering_device_vulkan.cpp:6449) - Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:22.211  7391  7433 E godot   :    at: draw_list_bind_uniform_set (drivers/vulkan/rendering_device_vulkan.cpp:7640) - Condition "!dl" is true.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!E" is true. Returning: TEXTURE_SAMPLES_1
10-10 00:35:22.211  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:22.211  7391  7433 E godot   :    at: draw_list_bind_uniform_set (drivers/vulkan/rendering_device_vulkan.cpp:7640) - Condition "!dl" is true.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:22.211  7391  7433 E godot   :    at: draw_list_bind_index_array (drivers/vulkan/rendering_device_vulkan.cpp:7713) - Condition "!dl" is true.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!framebuffer" is true. Returning: INVALID_ID
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Texture (binding: 0, index 0) is not a valid texture.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!framebuffer" is true. Returning: INVALID_ID
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.211  7391  7433 E godot   :    at: draw_list_end (drivers/vulkan/rendering_device_vulkan.cpp:8068) - Condition "!draw_list" is true.
10-10 00:35:22.211  7391  7433 E godot   : USER ERROR: Condition "!E" is true. Returning: TEXTURE_SAMPLES_1
10-10 00:35:22.212  7391  7433 E godot   :    at: draw_list_bind_uniform_set (drivers/vulkan/rendering_device_vulkan.cpp:7640) - Condition "!dl" is true.
10-10 00:35:22.212  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:22.212  7391  7433 E godot   :    at: _allocate_from_uniforms (./servers/rendering/renderer_rd/uniform_set_cache_rd.h:129) - Condition "rid.is_null()" is true. Returning: rid
10-10 00:35:22.212  7391  7433 E godot   :    at: draw_list_bind_uniform_set (drivers/vulkan/rendering_device_vulkan.cpp:7640) - Condition "!dl" is true.
10-10 00:35:22.212  7391  7433 E godot   :    at: draw_list_begin (drivers/vulkan/rendering_device_vulkan.cpp:7318) - Condition "!framebuffer" is true. Returning: INVALID_ID
10-10 00:35:22.212  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:22.212  7391  7433 E godot   : USER ERROR: Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:22.212  7391  7433 E godot   :    at: draw_list_bind_index_array (drivers/vulkan/rendering_device_vulkan.cpp:7713) - Condition "!dl" is true.
10-10 00:35:22.212  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:22.212  7391  7433 E godot   :    at: _generate_version (servers/rendering/renderer_rd/pipeline_cache_rd.cpp:60) - Condition "pipeline.is_null()" is true. Returning: RID()
10-10 00:35:22.212  7391  7433 E godot   :    at: uniform_set_create (drivers/vulkan/rendering_device_vulkan.cpp:5911) - Condition "!texture" is true. Returning: RID()
10-10 00:35:22.212  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.212  7391  7433 E godot   :    at: draw_list_draw (drivers/vulkan/rendering_device_vulkan.cpp:7765) - Condition "!dl" is true.
10-10 00:35:22.212  7391  7433 E godot   : USER ERROR: Condition "!E" is true. Returning: TEXTURE_SAMPLES_1
10-10 00:35:22.212  7391  7433 E godot   :    at: _generate_version (servers/rendering/renderer_rd/pipeline_cache_rd.cpp:60) - Condition "pipeline.is_null()" is true. Returning: RID()
10-10 00:35:22.212  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:22.213  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Immediate draw list is already inactive.
10-10 00:35:22.213  7391  7433 E godot   :    at: draw_list_draw (drivers/vulkan/rendering_device_vulkan.cpp:7765) - Condition "!dl" is true.
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Condition "!framebuffer" is true. Returning: INVALID_ID
10-10 00:35:22.213  7391  7433 E godot   :    at: framebuffer_format_get_texture_samples (drivers/vulkan/rendering_device_vulkan.cpp:4081) - Condition "!E" is true. Returning: TEXTURE_SAMPLES_1
10-10 00:35:22.213  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Condition "!framebuffer_formats.has(p_framebuffer_format)" is true. Returning: RID()
10-10 00:35:22.213  7391  7433 E godot   :    at: draw_list_bind_uniform_set (drivers/vulkan/rendering_device_vulkan.cpp:7640) - Condition "!dl" is true.
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Condition "pipeline.is_null()" is true. Returning: RID()
10-10 00:35:22.213  7391  7433 E godot   :    at: draw_list_bind_render_pipeline (drivers/vulkan/rendering_device_vulkan.cpp:7566) - Condition "!dl" is true.
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Texture (binding: 0, index 0) is not a valid texture.
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Condition "!dl" is true.
10-10 00:35:22.213  7391  7433 E godot   : USER ERROR: Condition "!framebuffer" is true. Returning: INVALID_ID
10-10 00:35:22.213  7391  7433 E godot   :    at: uniform_set_create (drivers/vulkan/rendering_device_vulkan.cpp:5911) - Condition "!texture" is true. Returning: RID()
10-10 00:35:22.214  7391  7433 E godot   :    at: draw_list_bind_render_pipeline (drivers/vulkan/rendering_device_vulkan.cpp:7566) - Condition "!dl" is true.
10-10 00:35:22.214  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
10-10 00:35:22.214  7391  7433 E godot   :    at: draw_list_end (drivers/vulkan/rendering_device_vulkan.cpp:8068) - Condition "!draw_list" is true.
```

repeat endlessly!!!

### Steps to reproduce

please use the reproduction project

### Minimal reproduction project

[demo.zip](https://github.com/godotengine/godot/files/9744635/demo.zip)

**Comment by Calinou** [comment]
Does this occur if you *only* have the OmniLight3D visible, or only have the SpotLight3D visible? The same goes for DirectionalLight3D.

**Comment by norrath-hero** [comment]
It's not Error spam, it's not working on Android, or because the endless spam app can't startup, so there only a black screen nothing else!

I will test more on this when I have time

**Comment by norrath-hero** [comment]
Here is my tests:

DirectionalLight3d:
  Shadow off, Nothing but a black screen
  Shadow on, Nothing but a black screen

OmniLight3d:
  Shadow off, Nothing but a black screen
  Shadow on, Nothing but a black screen

SpotLight3d:
  Shadow off, Nothing but a black screen
  Shadow on, Nothing but a black screen

then I remove all nodes except the root one:
   Nothing but a black screen

finally I change the root node to Node3D:
     Nothing but a black screen

it seems complete not working on my android device

**Comment by norrath-hero** [comment]
> > 10-10 00:35:21.848  7391  7433 E godot   : USER ERROR: vkCreateRenderPass2KHR failed with error -7.
> > 10-10 00:35:21.848  7391  7433 E godot   :    at: _render_pass_create (drivers/vulkan/rendering_device_vulkan.cpp:3969) - Condition "res" is true. Returning: nullptr
> 
> error code -7 is [VK_ERROR_EXTENSION_NOT_PRESENT](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VkResult.html). Your android 9 device might not support the [VK_KHR_create_renderpass2](https://registry.khronos.org/vulkan/specs/1.3-extensions/man/html/VK_KHR_create_renderpass2.html) extension, I guess.

Anything else can I do to help this? wait for beta3 then re-test?

**Comment by akien-mga** [comment]
Fixed by #67227.

**Comment by norrath-hero** [comment]
Just re-tested it on beta3 this time reveal a new issue
![image](https://user-images.githubusercontent.com/73905273/195973296-407ac989-7ed7-49fc-8b7f-05ebcd3329bd.png)
